include/mata/closed-set.hh

```diff
@@ -411,13 +414,13 @@ ClosedSet<T> ClosedSet<T>::complement() const {
         // The result will be an intersection of all downward-closed sets
         // created using this procedure.
         for(const auto& element : antichain()) {
-            ClosedSet preparingAntichain(downward_closed_set, get_min(), get_max());
-            for(int i = 0; i <= max_val_; ++i) {
+            ClosedSet preparingAntichain(ClosedSetType::downward_closed_set, get_min(), get_max());
+            for(int i = 0; i <= static_cast<int>(max_val_); ++i) {
                 if(!element.count(i)) {
                     continue;
                 }
                 Node candidates{};
-                for(int j = 0; j <= max_val_; ++j) {
+                for(int j = 0; j <= static_cast<int>(max_val_); ++j) {
```

Comment:
This only explicitly writes out what happens implicitly behind during compilation. The type conversion is executed regardless of whether you write it out explicitly or not.

Reply:
Therefore, this change did not affect performance. But as you suggest, we can optimize the further by extracting the cast outside the for loop manually.

Reply:
That is why we should listen to these warnings ;) Good catch.

Reply:
Fair enough. To be honest, I think it might have minimal impact (I'm integer type conversion should IMO be costless), it just stand out and extract might also make it more readable. Or maybe change the `i` to unsigned instead? Does it need to be singed?

Reply:
I will make the `j` variable of generic type `T` (the type of `max_val_`). That will resolve the issue for everywhere.

Reply:
Thank you for the discussion. This was actually a great optimization because we have to convert the `int j` again later into the `T` type anyway.

Reply:
Accidental optimization :D